src/DocumentStore.zig

```diff
@@ -588,6 +600,32 @@ fn getOrLoadBuildFile(self: *DocumentStore, uri: Uri) ?*BuildFile {
     return gop.value_ptr.*;
 }
 
+pub fn doubleCheckPotentialBuildFileUri(self: *DocumentStore, handle: *Handle) error{OutOfMemory}!?Uri {
+    handle.impl.lock.lock();
+    defer handle.impl.lock.unlock();
+    switch (handle.potential_build_file_uris.items.len) {
+        0 => {},
+        else => |count| blk: {
+            _ = self.getOrLoadBuildFile(handle.potential_build_file_uris.items[0]) orelse break :blk;
+            handle.associated_build_file = handle.potential_build_file_uris.items[0];
+            if (count == 1) break :blk;
+            const ok: bool = ok: for (handle.potential_build_file_uris.items[1..], 1..) |build_file_uri, index| {
+                const build_file = self.getOrLoadBuildFile(build_file_uri) orelse break :ok false;
+                if (!build_file.hasConfig()) break :ok false;
+                if (try self.uriAssociatedWithBuild(build_file, handle.uri)) {
+                    handle.associated_build_file = handle.potential_build_file_uris.items[index];
+                    break :ok true;
+                }
+            } else unreachable;
```

Comment:
Why is the else `unreachable`?

Reply:
The `for` is guaranteed at least one entry/iteration.

Reply:
How is the number of iterations related to whether or the not the else path is reached?

You code is equivalent to:
```zig
for (handle.potential_build_file_uris.items[1..], 1..) |build_file_uri, index| {
    const build_file = self.getOrLoadBuildFile(build_file_uri) orelse break :ok false;
    if (!build_file.hasConfig()) break :ok false;
    if (try self.uriAssociatedWithBuild(build_file, handle.uri)) {
        handle.associated_build_file = handle.potential_build_file_uris.items[index];
        break :ok true;
    }
    // if this code path here is reached on the last iteration the you hit unreachable.
    // I don't see why it shouldn't be possible to reach on the last iteration.
    // The handle may not be associated with any of the potential build files.
} else unreachable
```

Reply:
TIL; I thought it'd be only if the condition failed.